mne/source_space.py

```diff
@@ -2275,6 +2275,112 @@ def get_volume_labels_from_aseg(mgz_fname):
     return label_names
 
 
+def _get_hemi(s):
+    """Helper to get a hemisphere from a given source space"""
+    if s['type'] != 'surf':
+        raise RuntimeError('Only surface source spaces supported')
+    if s['id'] == FIFF.FIFFV_MNE_SURF_LEFT_HEMI:
+        return 'lh', 0
+    elif s['id'] == FIFF.FIFFV_MNE_SURF_RIGHT_HEMI:
+        return 'rh', 1
+    else:
+        raise ValueError('unknown surface ID %s' % s['id'])
+
+
+def _get_vertex_map_nn(src, morph, rrs):
+    """Helper to get a nearest-neigbor vertex match"""
+    # adapted from mne_make_source_space.c, knowing accurate=False (i.e.
+    # nearest-neighbor mode should be used)
+    morph_inuse = np.zeros(morph['np'], int)
+    best = np.zeros(len(src['vertno']), int)
+    ones = _compute_nearest(rrs[1], rrs[0][src['vertno']])
+    for v, one in zip(src['vertno'], ones):
+        # if it were actually a proper morph map, we would do this, but since
+        # we know it's nearest neighbor list, we don't need to:
+        # this_mm = mm[v]
+        # one = this_mm.indices[this_mm.data.argmax()]
+        if morph_inuse[one]:
+            # Try the nearest neighbors
+            neigh = _get_surf_neighbors(morph, one)  # on demand calc
+            was = one
+            one = neigh[np.where(~morph_inuse[neigh])[0]]
+            if len(one) == 0:
+                raise RuntimeError('vertex %d would be used multiple times.'
+                                   % one)
+            one = one[0]
+            logger.info('Source space vertex moved from %d to %d because of '
+                        'double occupation.' % (was, one))
+        best[v] = one
+        morph_inuse[one] = True
+    return best
+
+
+@verbose
+def morph_source_spaces(src_from, subject_to, surf='white', subjects_dir=None,
+                        verbose=None):
+    """Morph an existing source space to a different subject
+
+    .. note:: This can be used in place of morphing source estimates for
+              multiple subjects, but there may be consequences in terms
+              of dipole topology.
+
+    Parameters
+    ----------
+    src_from : instance of SourceSpaces
+        Surface source spaces to morph.
+    subject_to : str
+        The destination subject.
+    surf : str
+        The brain surface to use for the new source space.
```

Comment:
is it necessary to provide a surf param? is the output different if you use surf = 'pial' for example? I though it was just using the sphere reg to do the mapping.